src/combinatorics/quiver/single_obj.lean

```diff
@@ -0,0 +1,112 @@
+itself using `path_equiv_list`.
+-/
+
+namespace quiver
+
+/-- Type tag on `unit` used to define single-object quivers. -/
+@[nolint unused_arguments]
+def single_obj (α : Type*) : Type := unit
+
+namespace single_obj
+
+variables (α β γ : Type*)
+
+instance : quiver (single_obj α) := ⟨λ _ _, α⟩
+
+/-- The single object in `single_obj α`. -/
+def star : single_obj α := unit.star
+
+instance : inhabited (single_obj α) := ⟨star α⟩
+
+variables {α β γ}
+
+/-- Equip `single_obj α` with a reverse operation. -/
+@[reducible] -- See note [reducible non-instances]
+def has_reverse (rev : α → α) : has_reverse (single_obj α) := ⟨λ _ _, rev⟩
+
+/-- Equip `single_obj α` with an involutive reverse operation. -/
+@[reducible] -- See note [reducible non-instances]
+def has_involutive_reverse (rev : α → α) (h : function.involutive rev) :
+  has_involutive_reverse (single_obj α) :=
+{ to_has_reverse := has_reverse rev,
+  inv' := λ _ _, h}
+
+/-- The type of arrows from `star α` to itself is equivalent to the original type `α`. -/
+@[simps] def to_hom : α ≃ (star α ⟶ star α) := equiv.refl _
+
+/--
+Prefunctors between two `single_obj` quivers correspond to functions between the corresponding
+arrows types.
+-/
+def map_fun :
+  (α → β) ≃ (single_obj α ⥤q single_obj β) :=
+{ to_fun := λ f, ⟨id, λ _ _, f⟩,
+  inv_fun := λ f a, f.map (to_hom a),
+  left_inv := λ _, rfl,
+  right_inv :=  λ f, by cases f; obviously }
+
+lemma map_fun_id : map_fun id = 𝟭q (single_obj α) := rfl
+
+@[simp] lemma map_fun_symm_id :
+  map_fun.symm (𝟭q (single_obj α)) = id := rfl
+
+lemma map_fun_comp (f : α → β) (g : β → γ) :
+  map_fun (g ∘ f) = (map_fun f ⋙q map_fun g) := rfl
+
+@[simp] lemma map_fun_symm_comp (f : single_obj α ⥤q single_obj β)
+  (g : single_obj β ⥤q single_obj γ) : map_fun.symm (f ⋙q g) =
+  (map_fun.symm g ∘ map_fun.symm f) :=
+by simp only [equiv.symm_apply_eq, map_fun_comp, equiv.apply_symm_apply]
+
+/--
+Auxiliary definition for `quiver.single_obj.path_equiv_list`.
+Converts a path in the quiver `single_obj α` into a list of elements of type `a`.
+-/
+@[simp] def path_to_list : Π {x : single_obj α}, path (star α) x → list α
+| _ path.nil := []
+| _ (path.cons p a) := a :: (path_to_list p)
+
+/--
+Auxiliary definition for `quiver.single_obj.path_equiv_list`.
+Converts a list of elements of type `α` into a path in the quiver `single_obj α`.
+-/
+@[simp] def list_to_path : list α → path (star α) (star α)
+| [] := path.nil
+| (a :: l) := (list_to_path l).cons a
+
+lemma path_to_list_to_path {x : single_obj α} (p : path (star α) x) :
+  list_to_path (path_to_list p) = p.cast rfl (unit.ext) :=
+by { induction p with y z p a ih, refl, tidy, }
+
+lemma list_to_path_to_list (l : list α) :
+  path_to_list (list_to_path l) = l :=
+by { induction l with a l ih, refl, simp [ih] }
+
+/-- Paths in `single_obj α` quiver correspond to lists of elements of type `α`. -/
+@[simps] def path_equiv_list : path (star α) (star α) ≃ list α :=
```

Comment:
Either you consider `path_to_list` and `list_to_path` as auxiliary definitions that shoudn't be exposed
```suggestion
def path_equiv_list : path (star α) (star α) ≃ list α :=
```
and you add the simp lemmas for `path_equiv_list path.nil`, `path_equiv_list (path.cons a p)`, `path_equiv_list.symm []`, `path_equiv_list.symm (a :: l)`

Or you consider them proper definitions, in which case you should remove "Auxiliary" from their docstrings to reflect that.